actions/pull-request/check/entrypoint

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+set -eu
+set -o pipefail
+
+function main() {
+  local token repo number botsString
+
+  while [ "${#}" != 0 ]; do
+    case "${1}" in
+      --token)
+        token="${2}"
+        shift 2
+        ;;
+
+      --repo)
+        repo="${2}"
+        shift 2
+        ;;
+
+      --number)
+        number="${2}"
+        shift 2
+        ;;
+
+      --bots)
+        botsString="${2}"
+        shift 2
+        ;;
+
+      "")
+        shift
+        ;;
+
+      *)
+        echo "unknown argument \"${1}\""
+        exit 1
+    esac
+  done
+
+  if rules::has_human_commits "${token}" "${repo}" "${number}" "${botsString}"; then
+    echo "::set-output name=human_commits::true"
+    exit 0
+  fi
+    echo "::set-output name=human_commits::false"
+}
+
+function rules::has_human_commits() {
+  local token repo number bots humanCommit
+  token=${1}
+  repo=${2}
+  number=${3}
+  botsString=${4}
+
+  IFS="," read -r -a bots <<< "${botsString}"
+
+  # this allows us to parse a newline-separated set of names (which might
+  # include spaces) that come out of jq and iterate over them without
+  # splitting on spaces
+  while IFS= read -r author; do
+      humanCommit=0
+      for bot in "${bots[@]}"; do
+        if [[ "${author}" == "${bot}" ]]; then
+          humanCommit=1
+          break
+        fi
+      done
+
+      if [[ ${humanCommit} -eq 0 ]]; then
+        echo "human committer: " "${author}"
+        return 0
+      fi
+  done < <(curl "https://api.github.com/repos/${repo}/pulls/${number}/commits" \
+    --silent \
+    --location \
+    --header \
+    "Authorization: token ${token}" | jq -r '.[] | .commit.author.name, .commit.committer.name')
```

Comment:
Why are we using the `name` fields on the commit rather than the `login` fields under `committer` and `author`?

Reply:
We have discovered that it is possible for the `login` fields under `committer` and `author` to contain `null` in some cases ([example](https://github.com/paketo-buildpacks/dotnet-core-runtime/pull/138/commits/083c0227915c313d15065bbd6209af2aaf79c71a)). We considered it safer and less confusing to use the `name` field rather than perhaps adding `null` as an element in `bots`.

Reply:
I think this is caused by us using the wrong email when creating commits here: https://github.com/paketo-buildpacks/github-config/blob/509aa0b6e02d46d1a2430e7a2cc92a53335aee7b/actions/sync/entrypoint#L144. I'm going to fix that. I worry that if we use the name of the committer, there isn't anything stopping someone from changing their commit signature name to `paketo-bot` and getting past this check. We'd also want to ensure the commits that come from these bots are signed. That's the only way you can truly verify that the committer is who they say they are.